src/marshmallow_sqlalchemy/convert.py

```diff
@@ -35,6 +35,24 @@ def _postgres_array_factory(converter, data_type):
     )
 
 
+def _field_update_kwargs(field_class, field_kwargs, kwargs):
+    if not kwargs:
+        return
+    # set of fields possible keyword arguments
+    possible_field_keywords = {
+        key
+        for cls in inspect.getmro(field_class)
+        for key, param in inspect.signature(cls).parameters.items()
+        if param.kind.description == "positional or keyword"
+        or param.kind.description == "keyword-only"
+    }
+    for k, v in kwargs.items():
+        if k in possible_field_keywords:
+            field_kwargs[k] = v
+        else:
+            field_kwargs["metadata"][k] = v
+
+
```

Comment:
In order to bucket non-field arguments in to a `metadata` argument, we need to be able to differentiate which kwargs to `property2field()` and `column2field()` are field parameters and which aren't.

This creates a set of the names of each of the potential keyword parameters of the field class and its parents and then checks if each of the kwargs is a part of that set, or not, and allocates to `field_kwargs` or `field_kwargs["metadata"]` as appropriate.